lib/ZwaveClient.ts

```diff
@@ -961,6 +980,173 @@ class ZwaveClient extends TypedEventEmitter<ZwaveClientEventCallbacks> {
 		}
 	}
 
+	/**
+	 * If the node supports Schedule Lock CC parses all available schedules and cache them
+	 */
+	async getSchedules(nodeId: number) {
+		const zwaveNode = this.getNode(nodeId)
+
+		if (!zwaveNode?.commandClasses['Schedule Entry Lock'].isSupported()) {
+			throw new Error(
+				'Schedule Entry Lock CC not supported on node ' + nodeId
+			)
+		}
+
+		const userCodes = await zwaveNode.commandClasses[
+			'User Code'
+		].getUsersCount()
+
+		const numSlots = await zwaveNode.commandClasses[
+			'Schedule Entry Lock'
+		].getNumSlots()
```

Comment:
This will request the slots from the lock again. This information **should** be queried during the interview, so you can read the cached information using
```ts
ScheduleEntryLockCC.getNumWeekDaySlotsCached(driver, zwaveNode)
```
and similar for the other types.

https://github.com/zwave-js/node-zwave-js/blob/4854b0cf2b2c9bad15ac1b42a2632776e28db4b7/packages/cc/src/cc/ScheduleEntryLockCC.ts#L492-L543

Reply:
Argument of type 'Driver' is not assignable to parameter of type 'ZWaveApplicationHost'.
  Type 'Driver' is missing the following properties from type 'ZWaveApplicationHost': options, controllerLog, getValueDB, tryGetValueDB, and 10 more